types/authorizationRequest.ts

```diff
@@ -0,0 +1,129 @@
+import {Relationship} from "./common"
+
+export interface PurchaseAuthorizationRequest {
+    /**
+     * Identifier of the authorization resource.
+     */
+    id: string
+
+    /**
+     * Type of the authorization resource. The value is always authorization.
+     */
+    type: "purchaseAuthorizationRequest"
+
+    /**
+     * JSON object representing the authorization data.
+     */
+    attributes: {
+        /**
+         * RFC3339 Date string	The date the authorization was created.
+         */
+        createdAt: string
+
+        /**
+         * The amount (cents) of the authorization.
+         */
+        amount: number
+
+        /**
+         * The status of the authorization request. Either Pending, Approved or Declined.
+         */
+        status: string
+
+        /**
+         * Indicates whether the authorization request supports partial amount approval.
+         */
+        partialApprovalAllowed: boolean
+
+        /**
+         * Optional. The amount (cents) that was approved. Available only when status is Approved.
+         */
+        approvedAmount?: number
+
+        /**
+         * 	Optional. The reason the authorization request was declined.
+         * 	One of  AccountClosed,
+         * 	        CardExceedsAmountLimit,
+         * 	        DoNotHonor,
+         * 	        InsufficientFunds,
+         * 	        InvalidMerchant,
+         * 	        ReferToCardIssuer,
+         * 	        RestrictedCard,
+         * 	        Timeout,
+         * 	        TransactionNotPermittedToCardholder.
+         * Available only when status is Declined
+         */
+        declineReason?: string
```

Comment:
we are using this convention

```suggestion
        declineReason?:  "AccountClosed" | "CardExceedsAmountLimit" | ....
```

Reply:
thank you on your quick response, i will do it